community/contributing/bisect_workflow.rst

```diff
@@ -0,0 +1,131 @@
+
+The guide below explains what's bisecting and how to find a regression
+by bisecting.
+
+What is bisecting?
+------------------
+
+Godot developers use the `Git <https://git-scm.com/>`__ version control system.
+In the context of Git, bisecting is the process of performing a manual
+`binary search <https://en.wikipedia.org/wiki/Binary_search_algorithm>`__
+to determine when a regression appeared. While it's typically used for bugs,
+it can also be used to find other kinds of unexpected changes such as
+performance regressions.
+
+Using official builds to speed up bisecting
+-------------------------------------------
+
+Before using Git's ``bisect`` command, we strongly recommend trying to reproduce
+the bug with an older (or newer) official release. This greatly reduces the
+range of commits that potentially need to be built from source and tested. You
+can find binaries of official releases, as well as alphas, betas, and release
+candidates `here <https://downloads.tuxfamily.org/godotengine/>`__.
+
+For example, if you've reported a bug against Godot 3.2 RC4, you should first
+try to reproduce the bug in Godot 3.1.2. If the bug doesn't occur there, try to
+reproduce it in Godot 3.2 alpha 1. If the bug doesn't occur there either, the
+regression appeared between Godot 3.1.2 and 3.2 alpha 1. Otherwise, it appeared
+in a newer pre-release build; you should now test newer alphas, betas and RC
+builds.
+
+The Git bisect command
+----------------------
+
+If you've found a build that didn't exhibit the bug in the above testing
+process, you can now start bisecting the regression. The Git version control
+system offers a built-in command for this: ``git bisect``. This makes the
+process semi-automated as you only have to build the engine, run it and try to
+reproduce the bug.
+
+.. note::
+
+    Before bisecting a regression, you need to set up a build environment to
+    compile Godot from source. To do so, read the
+    :ref:`Compiling <toc-devel-compiling>` page for your target platform.
+    (Compiling Godot from source doesn't require C++ programming knowledge.)
+
+To start bisecting, you must first determine the commit hashes (identifiers) of
+the "bad" and "good" build. "bad" refers to the build that exhibits the bug,
+whereas "good" refers to the version that doesn't exhibit the bug. If you're
+using a pre-release build as the "good" or "bad" build, browse the `download
+mirror <https://downloads.tuxfamily.org/godotengine/>`__, go to the folder that
+contains the pre-release you downloaded and look for the ``README.txt`` file.
+The commit hash is written inside that file.
+
+If you're using a stable release as the "good" or "bad" build, use one of the
+following commit hashes depending on the version:
+
+.. code-block:: none
+
+    3.1.2: 0587df4aa5f2977350cc80b1522cdc1e483c4515
+    3.1.1: 66baa3b633fe904ea0d90a9688d602d9f3a0b3bd
+    3.1: 320f49f204cfbf9b480fe62aaa7718afb74920a5
+    3.0.6: 8ac39d886307d76c286e804e027fc39f6b5aaac6
```

Comment:
If you are bisecting between 3.0 and 3.1, then it's not useful to have the hash of 3.0.6, since 3.1 and 3.0.6 are on separate branches and their common ancestor is 3.0.

If you are bisecting between 3.0 and 3.0.6, then it's also not useful to have the hash of 3.0.6, since you can just start from the head of the 3.0 branch.

So, I think this article should only mention the hashes of unpatched releases, right now that would mean only 3.0 and 3.1.